src/linear_algebra/determinant.lean

```diff
@@ -283,12 +283,8 @@ begin
     rintros ⟨k, x⟩,
     simp },
   { intros σ _,
-    rw finset.prod_mul_distrib,
-    congr,
-    { convert congr_arg (λ (x : units ℤ), (↑x : R)) (sign_prod_congr_left (λ k, σ k _)).symm,
-      simp, congr, ext, congr },
-    rw [← finset.univ_product_univ, finset.prod_product, finset.prod_comm],
-    simp },
+    rw [finset.prod_mul_distrib, ←finset.univ_product_univ, finset.prod_product, finset.prod_comm],
+    simp [sign_prod_congr_left] },
```

Comment:
This is a great improvement!

I wasn't aware that using `classical` like this could be an issue. Are there other places where this pops up?

Reply:
Yes, the `sign` lemmas simplifying various `congr` were causing me a lot of pain trying to simplify pretty obvious `equiv_congr.map_equiv option`, because the `decidable_eq` instance for `option α` is easily constructed from the `[decidable_eq α]`, but the relevant lemma had no explicit `decidable_eq`! I was so confused, until I noticed the `open classical` that was near the end of the file. Some new `sign` lemmas didn't have this issue because they used different `Type` arguments.

Reply:
I found an earlier Zulip thread that discussed a similar issue and mentioned this PR on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/finset.20coercion.20strangeness/near/221728442).